modules/client/test/integration/client/encoding.test.ts

```diff
@@ -610,50 +605,200 @@ describe("Client", () => {
+        }
+      }
+    });
+    it("Should encode an applyUpdate action with permissions", async () => {
+      const context = new Context(contextParamsLocalChain);
+      const client = new Client(context);
 
-      const daoInterface = PluginSetupProcessor__factory.createInterface();
-      const hexString = bytesToHex(actions[1].data);
-      const argsDecoded = daoInterface.decodeFunctionData(
-        "applyUpdate",
-        hexString,
-      );
-      expect(argsDecoded.length).toBe(2);
-      expect(argsDecoded[0]).toBe(
+      const daoAddress = ADDRESS_ONE;
+      const pluginAddress = ADDRESS_TWO;
+      const applyUpdateParams: ApplyUpdateParams = {
+        permissions: [{
+          operation: 1,
+          permissionId: PermissionIds.EXECUTE_PERMISSION_ID,
+          where: "0x1234567890123456789012345678901234567890",
+          who: "0x2345678901234567890123456789012345678901",
+        }],
+        versionTag: {
+          build: 1,
+          release: 1,
+        },
+        pluginRepo: "0x2345678901234567890123456789012345678901",
+        pluginAddress,
+        initData: new Uint8Array([0, 1, 2, 3]),
+        helpers: [],
+      };
+      const actions = client.encoding.applyUpdateAction(
         daoAddress,
+        applyUpdateParams,
       );
-      expect(argsDecoded[1].pluginSetupRef.versionTag.build).toBe(
-        applyUpdateParams.versionTag.build,
-      );
-      expect(argsDecoded[1].pluginSetupRef.versionTag.release).toBe(
-        applyUpdateParams.versionTag.release,
-      );
-      expect(argsDecoded[1].plugin).toBe(
-        applyUpdateParams.pluginAddress,
-      );
-      expect(argsDecoded[1].pluginSetupRef.pluginSetupRepo).toBe(
-        applyUpdateParams.pluginRepo,
-      );
-      for (const index in argsDecoded[1].permissions) {
-        expect(argsDecoded[1].permissions[parseInt(index)].operation).toBe(
-          applyUpdateParams.permissions[parseInt(index)].operation,
-        );
-        expect(argsDecoded[1].permissions[parseInt(index)].where).toBe(
-          applyUpdateParams.permissions[parseInt(index)].where,
-        );
-        expect(argsDecoded[1].permissions[parseInt(index)].who).toBe(
-          applyUpdateParams.permissions[parseInt(index)].who,
-        );
-        expect(argsDecoded[1].permissions[parseInt(index)].condition).toBe(
-          AddressZero,
-        );
-        expect(argsDecoded[1].permissions[parseInt(index)].permissionId).toBe(
-          applyUpdateParams.permissions[parseInt(index)]
-            .permissionId,
-        );
+      const pspInterface = PluginSetupProcessor__factory.createInterface();
+      const daoInterface = DAO__factory.createInterface();
+      const expectedActions = ["grant", "grant", "applyUpdate", "revoke", "revoke"];
+      let hexString, argsDecoded, action;
+      expect(actions.length).toBe(5);
+      for (const [index, actionName] of expectedActions.entries()) {
+        action = actions[index];
```

Comment:
Fix order. Why are case 0 and 1 missing?

Reply:
the order is correct, same as on the other comment its the JS way of saying `case 'grant' || 'revoke'`